solr/core/src/test/org/apache/solr/cloud/CollectionsAPIDistributedZkTest.java

```diff
@@ -70,357 +77,193 @@
-
-  private static final String DEFAULT_COLLECTION = "collection1";
+  
+  private static final int NODE_COUNT = 5;
 
   // we randomly use a second config set rather than just one
-  private boolean secondConfigSet = random().nextBoolean();
+  private final boolean secondConfigSet = false; //random().nextBoolean(); FIXME
+  private final String configSet = secondConfigSet ? "conf1": "conf2";
+
+  private static CloudSolrClient cloudClient;
+  private static ZkTestServer zkServer;
   
   @BeforeClass
-  public static void beforeCollectionsAPIDistributedZkTest() {
+  public static void setup() throws Exception {
+    configureCluster(NODE_COUNT)
+        .addConfig("conf1", TEST_PATH().resolve("configsets").resolve("cloud-jmx").resolve("conf"))
+        .addConfig("conf2", TEST_PATH().resolve("configsets").resolve("cloud-jmx").resolve("conf"))
+        .configure();
+    
+    cloudClient = cluster.buildSolrClient(random().nextBoolean());
+    cloudClient.setParallelUpdates(random().nextBoolean());
+    cloudClient.connect();
+    
+    zkServer = cluster.getZkServer();
+    
     TestInjection.randomDelayInCoreCreation = "true:20";
     System.setProperty("validateAfterInactivity", "200");
   }
-  
-  @Override
-  public void distribSetUp() throws Exception {
-    super.distribSetUp();
-    
-    if (secondConfigSet ) {
-      String zkHost = zkServer.getZkHost();
-      String zkAddress = zkServer.getZkAddress();
-      SolrZkClient zkClient = new SolrZkClient(zkHost, AbstractZkTestCase.TIMEOUT);
-      zkClient.makePath("/solr", false, true);
-      zkClient.close();
-
-      zkClient = new SolrZkClient(zkAddress, AbstractZkTestCase.TIMEOUT);
 
-      File solrhome = new File(TEST_HOME());
-      
-      // for now, always upload the config and schema to the canonical names
-      AbstractZkTestCase.putConfig("conf2", zkClient, solrhome, getCloudSolrConfig(), "solrconfig.xml");
-      AbstractZkTestCase.putConfig("conf2", zkClient, solrhome, "schema.xml", "schema.xml");
-      AbstractZkTestCase.putConfig("conf2", zkClient, solrhome, "enumsConfig.xml", "enumsConfig.xml");
-      
-      AbstractZkTestCase.putConfig("conf2", zkClient, solrhome, "solrconfig.snippet.randomindexconfig.xml");
-      AbstractZkTestCase.putConfig("conf2", zkClient, solrhome, "stopwords.txt");
-      AbstractZkTestCase.putConfig("conf2", zkClient, solrhome, "protwords.txt");
-      AbstractZkTestCase.putConfig("conf2", zkClient, solrhome, "currency.xml");
-      AbstractZkTestCase.putConfig("conf2", zkClient, solrhome, "open-exchange-rates.json");
-      AbstractZkTestCase.putConfig("conf2", zkClient, solrhome, "mapping-ISOLatin1Accent.txt");
-      AbstractZkTestCase.putConfig("conf2", zkClient, solrhome, "old_synonyms.txt");
-      AbstractZkTestCase.putConfig("conf2", zkClient, solrhome, "synonyms.txt");
-      AbstractZkTestCase.putConfig("conf2", zkClient, solrhome, "elevate.xml");
-      zkClient.close();
-    }
+  @AfterClass
```

Comment:
Actually, I at least need to close() the zkServer, as this (50% of the time) gets replaced by a new one when restarting Zookeeper.